subgraphs/diva-staking/mappings/ComptrollerLib.ts

```diff
@@ -0,0 +1,60 @@
+import { createDeposit } from '../entities/Deposit';
+import { SharesBought, SharesRedeemed } from '../generated/contracts/ComptrollerLibEvents';
+import { calcVaultsGav } from '../utils/calcVaultsGav';
+import {
+  getAccruedRewards,
+  getDepositTranches,
+  getRedemptionTranches,
+  stakingDeadlineTimestamp,
+} from '../utils/tranches';
+import { getDepositor, updateDepositor, createDepositor } from '../entities/Depositor';
+import { createRedemption } from '../entities/Redemption';
+import { store } from '@graphprotocol/graph-ts';
+import { updateDepositorCounter } from '../entities/GeneralInfo';
+
+export function handleSharesBought(event: SharesBought): void {
+  // skip deposits after staking deadline
+  if (stakingDeadlineTimestamp < event.block.timestamp) {
+    return;
+  }
+
+  let vaultsGav = calcVaultsGav();
+  let investmentAmount = event.params.investmentAmount;
+  let vaultsGavBeforeDeposit = vaultsGav.minus(investmentAmount);
+
+  let tranches = getDepositTranches(vaultsGavBeforeDeposit, investmentAmount);
+
+  createDeposit(event.params.buyer, tranches, event);
+
+  let depositor = getDepositor(event.params.buyer);
+
+  if (depositor) {
+    updateDepositor(depositor, tranches, event.params.sharesReceived, event);
+  } else {
+    createDepositor(event.params.buyer, tranches, event.params.sharesReceived, event);
+    updateDepositorCounter(1, event); // increase by 1
+  }
+}
+
+export function handleSharesRedeemed(event: SharesRedeemed): void {
+  let depositor = getDepositor(event.params.redeemer);
+
+  if (depositor == null) {
+    return;
+  }
```

Comment:
Create a function `useDepositor`, like we do in many other places. That function should throw if the depositor is not found.